src/Aspire.Hosting.Azure/AzureDeployingContext.cs

```diff
@@ -61,43 +60,27 @@ public async Task DeployModelAsync(AzureEnvironmentResource resource, Distribute
         }
     }
 
-    private async Task<bool> TryProvisionAzureInfrastructure(AzureEnvironmentResource resource, ProvisioningContext provisioningContext, CancellationToken cancellationToken)
+    private async Task<bool> TryProvisionAzureBicepResources(List<AzureBicepResource> bicepResources, ProvisioningContext provisioningContext, CancellationToken cancellationToken)
     {
-        var deployingStep = await activityReporter.CreateStepAsync("Deploying to Azure", cancellationToken).ConfigureAwait(false);
+        var deployingStep = await activityReporter.CreateStepAsync("Deploying Azure resources", cancellationToken).ConfigureAwait(false);
         await using (deployingStep.ConfigureAwait(false))
         {
             try
             {
-                foreach (var (parameterResource, provisioningParameter) in resource.PublishingContext!.ParameterLookup)
+                var deploymentGraph = ResourceDeploymentGraph.Build(bicepResources);
```

Comment:
The way it works for `dotnet run` is that a resource that has a BicepOutputReference blocks until that BicepOutputReference is filled with a value. So there is no need to build up a "graph". You just start deploying all resources, and the graph "just works".

Can that model be made to work here?

Reply:
Yes the implicit graph model works. The downside is that cycles result in a hang😄. Whatever we decide, I'd like to unify run and deploy.